packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFScrollViewDelegateHostApi.m

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FWFScrollViewDelegateHostApi.h"
+#import "FWFWebViewHostApi.h"
+
+@interface FWFScrollViewDelegateFlutterApiImpl ()
+// BinaryMessenger must be weak to prevent a circular reference with the host API it
+// references.
+@property(nonatomic, weak) id<FlutterBinaryMessenger> binaryMessenger;
+// InstanceManager must be weak to prevent a circular reference with the object it stores.
+@property(nonatomic, weak) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFScrollViewDelegateFlutterApiImpl
+
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self initWithBinaryMessenger:binaryMessenger];
+  if (self) {
+    _binaryMessenger = binaryMessenger;
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+- (long)identifierForDelegate:(FWFScrollViewDelegate *)instance {
+  return [self.instanceManager identifierWithStrongReferenceForInstance:instance];
+}
+
+- (void)scrollViewDidScrollForDelegate:(FWFScrollViewDelegate *)instance
+                          uiScrollView:(UIScrollView *)scrollView
+                            completion:(void (^)(FlutterError *_Nullable))completion {
+  [self scrollViewDidScrollWithIdentifier:[self identifierForDelegate:instance]
+                   uiScrollViewIdentifier:[self.instanceManager
+                                              identifierWithStrongReferenceForInstance:scrollView]
+                                        x:scrollView.contentOffset.x
+                                        y:scrollView.contentOffset.y
+                               completion:completion];
+}
+@end
+
+@implementation FWFScrollViewDelegate
+
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [super initWithBinaryMessenger:binaryMessenger instanceManager:instanceManager];
+  if (self) {
+    _scrollViewDelegateAPI =
+        [[FWFScrollViewDelegateFlutterApiImpl alloc] initWithBinaryMessenger:binaryMessenger
+                                                             instanceManager:instanceManager];
```

Comment:
this seems to be just a 1-line wrapper. can `FWFScrollViewDelegate` itself conform to `FWFScrollViewDelegateFlutterApi`?

Reply:
I believe it's possible, but I adhered to the conventions established by the previously implemented API, such as the FWFUIDelegate.

Reply:
The reason this pattern doesn't have the wrapped class implement their own FlutterApi is because it wouldn't work well with inheritance. For example if we had

```dart
class A { }
class B implements A { }

interface AFlutterApi { } 
interface BFlutterApi { }
```

`class B` would need access to an implementation of `AFlutterApi` and `BFlutterApi`. So, there wouldn't be an easy way to do this if we did `class A implements AFlutterApi`. 

Note that this will also be solved by https://github.com/flutter/flutter/issues/134777 by handling inherited flutter api automatically.